theforeman.org/scripts/debian/stage_packages.sh

```diff
@@ -7,19 +7,44 @@ set -xe
 # * the web node must have the freight class applied
 
 # Find the deps in the build-dir from the previous step
-DEB_PATH=./debian/${os}/build-${project}
+DEB_PATH=$(find ./ -name build-${project})
```

Comment:
Is there no risk that it finds multiple projects now? Do we properly clean up? Also, can't we solve this with a simple glob? If anything, we can add a min and max depth so it doesn't do a very slow and fully recursive search. If there are a lot of files this can be very slow.

Reply:
I do not follow what the suggestion really is here. I can say that I think the risks are low as this is performed within a git checkout, and that checkout will be unique to the job run.

Reply:
I think there is no risk in different *projects*, as the find has the project in the search criteria. But different OS'es *could* clash (I think Jenkins *does* different checkouts, but we can be safe regardless).

The old logic was:
* core: `DEB_PATH=./debian/${os}/build-${project}`
* plugins: ` DEB_PATH=./plugins/build-${project}`
* deps: `DEB_PATH=./dependencies/${os}/build-${project}`

And the problem you're trying to solve is the fact that plugins doesn't have a `${os}` subdir?

The a tad verbose, but safe solution would be:
```bash
for possible_path in ./debian/${os}/build-${project} ./plugins/build-${project} ./dependencies/${os}/build-${project}; do
  if [[ -d "${possible_path}" ]]; then
    DEB_PATH="${possible_path}"
  fi
done
```

The less vebose would be `find . -mindepth 2 -maxdepth 3 -type d -name build-${project}`